intranet/static/js/bus-afternoon.js

```diff
@@ -95,23 +95,32 @@ $(function() {
             let busList = [];
             if (action === 'Search for a bus') {
                 busList = routeList.filter(bus => bus.attributes.status === 'a')
-                                    .filter(bus => bus.attributes.route_name.includes('JT'))
-                                    .map(bus => bus.attributes);
+                    .filter(bus => bus.attributes.route_name.includes('JT'))
+                    .map(bus => bus.attributes);
             } else if (action === 'Mark a bus as arrived or on time') {
-                busList = routeList.filter(bus => !bus.attributes.route_name.includes('JT'))
-                                    .map(bus => {
-                                        if (bus.attributes.status === 'a') {
-                                            // TODO: less hacky deep copy
-                                            let attr = JSON.parse(JSON.stringify(bus.attributes));
-                                            attr.route_name = `Mark ${bus.attributes.route_name} as on time`;
-                                            return attr;
-                                        } else {
-                                            return bus.attributes;
-                                        }
-                                    });
+                busList = routeList.filter(bus => bus.attributes.route_name.includes(''))
+                    .map(bus => {
+                        if (bus.attributes.status === 'a' || bus.attributes.status === 'd') {
+                            // TODO: less hacky deep copy
+                            let attr = JSON.parse(JSON.stringify(bus.attributes));
+                            attr.route_name = `Mark ${bus.attributes.route_name} as on time`;
```

Comment:
Fairfax buses (JT-###) should not have a "Mark as on time" option. Marking them on time should be done by unassigning them from their existing bus spot.